### PR TITLE
Improve permissions

### LINKED ISSUE #380: Problem with imports
Some "requires" definitions are incorrect. For example:
```JavaScript
Ext.define('Traccar.view.DeviceDialog', {
    extend: 'Traccar.view.BaseEditDialog',

    requires: [
        'Traccar.view.BaseEditDialog'
    ],

    controller: 'baseEditDialog',
```

In this case, instead of importing `BaseEditDialog` which is already imported via `extend`, we should import `BaseEditDialogController` which is required for `baseEditDialog` controller.

I guess we should go through the code and find if there are any other similar issues.

### LINKED ISSUE #387: Token is not fully visible
Because token field is disabled for a regular user, he can't read the full token value. We need to fix it somehow.

### PR DESCRIPTION
**Pull request by Abyss777**
Adjusted web interface to https://github.com/tananaev/traccar/issues/2810 https://github.com/tananaev/traccar/pull/2831 https://github.com/tananaev/traccar/pull/2835

User from admin account
image

New user from manager account
image

Ordinary user account (I've decided not hide "Device readonly" field, just disable)
image
 
Readonly view
image

Device readonly view
image

fix #387 

Also made up imports, fix #380

### CODE REVIEW

File: web/app/view/DevicesController.js [hunk 1]
```diff
@@ -87,7 +89,12 @@ Ext.define('Traccar.view.DevicesController', {
                 if (btn === 'yes') {
                     store = Ext.getStore('Devices');
                     store.remove(device);
-                    store.sync();
+                    store.sync({
+                        failure: function (batch) {
```
Comment (Abyss777) [inline web/app/view/DevicesController.js]: Show errors if device deleting failed

File: web/app/view/UsersController.js [hunk 2]
```diff
@@ -40,6 +40,12 @@ Ext.define('Traccar.view.UsersController', {
     onAddClick: function () {
         var user, dialog;
         user = Ext.create('Traccar.model.User');
+        if (Traccar.app.getUser().get('admin')) {
+            user.set('deviceLimit', -1);
```
Comment (Abyss777) [inline web/app/view/UsersController.js]: Added just not break old experiences, new user created admin will have infinite device limit by default

File: web/app/view/UsersController.js [hunk 3]
```diff
@@ -40,6 +40,12 @@ Ext.define('Traccar.view.UsersController', {
...
+        }
+        if (Traccar.app.getUser().get('expirationTime')) {
+            user.set('expirationTime', Traccar.app.getUser().get('expirationTime'));
```
Comment (Abyss777) [inline web/app/view/UsersController.js]: Made for convenience, if manager has expiration time, new user will have same expiration time by default.

**Comment by Abyss777** [comment]
Added small fix to determine manager correctly

File: web/app/view/DevicesController.js (hunk 1 above)
Comment (tananaev) [inline web/app/view/DevicesController.js]: Should we handle it for other service calls as well? Then I guess it should be shared somewhere.

Comment (Abyss777) [inline web/app/view/DevicesController.js]: Sure. It's not handled in some other places. I'll we check what we can do.

**Comment by djbo1982** [comment]
@Abyss777  @tananaev 
Hello man, im reading about one thing in other develop and i find this document, maybe is useful for user manager

copy as original 

I created a manager and give 10 device permission.
I then logged in with the manager and created a sub-manager called the manager and give 10 device access.
I went to admin and then tried to look at the user relations
The relation does not specify or show the which manager the sub-manager belongs

Problem.


[Relational.User.manager.pdf](https://github.com/tananaev/traccar-web/files/759712/Relational.User.manager.pdf)

**Comment by Abyss777** [comment]
@djbo1982 
Looks like it is not about official traccar. 
In the current implementation manager can not create manager.

**Pull request merged**